### DISCUSSION TITLE
[Implemented] Ability to include files inside customAlias.yml

### DISCUSSION THREAD

**Discussion by mrfloris** [Zrips/CMI#3791] [Feature Requests]
You know what would be cool, a way like PHP to include another `yml` file inside `customAlias.yml`. 

The ability to make/use a `commands/` directory where we can put our own files like `staff.yml`, `players.yml`, `temptest.yml`, `valentines.yml`, `specialevent.yml` etc. 

And in the `customalias.yml` file we could add at the bottom something like:
```
{include: commands/staff.yml}
{include: commands/players.yml}
{include: commands/specialevent.yml}
```

This would allow us to keep files with commands on record for future events, and to hot swap them in-game with a `/cmi reload` Without risking potentially accidentally deleting anything or the wrong commands. 

It would also allow us to share custom commands with others like `social-commands.yml` (random example)

[edit]

I guess you could even consider revamping things slightly. Where Alias.yml has all true by default. And people can still customize it, or keep what they have now. And to false it all, remove the require line from new `commands.yml`
```
commands.yml:
{require: alias.yml} (if not found, they default to false on all}
{include: customAlias.yml} (how it is now, for backwards compatibility)
{include: commands/*.yml} (tries to parse all .yml files matching regex)
~ or ~
{include: commands/event.yml}
{include: commands/staff.yml}
```
Then server owners don't even have to edit existing .yml files, and if they have already it won't be any different. 
And custom .yml files like staff or social or something customers share on discord, they can be added to commands.yml

**Comment by CloudeLecaw** [comment]
But the customalias is no loader for other files dont you think?
This including of other files sounds more like a feature that could be used everywhere in cmi but not only in custom alias. Just using it there would be a big limiatuin of  the things you could do with it

**Comment by mrfloris** [comment]
Please read things carefully.

**Comment by mrfloris** [comment]
This has been implemented, currently, the system isn't as suggested, but even easier. Easier for sharing that is.

You now have a CustomAlias/ directory where files.yml can be dropped - and they will be read in on server start and with /cmi reload.

**Reply by CloudeLecaw** [reply]
correct :D
